### PR TITLE
IR printer: for prim::Constant nodes with Function type print out the function name in the comment.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24383 IR printer: for prim::Constant nodes with Function type print out the function name in the comment.**

Differential Revision: [D16823850](https://our.internmc.facebook.com/intern/diff/D16823850)

### CODE REVIEW

**Review by suo — APPROVED** [review]
ngnt

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -258,6 +259,12 @@ std::ostream& Node::print(
       out << " # " << filename << ":" << line << ":" << col;
     }
   }
+  if (kind() == prim::Constant &&
+      output()->type()->kind() == FunctionType::Kind) {
+    auto funcname =
+        output()->type()->expect<FunctionType>()->function()->name();
+    out << " # " << funcname;
```
Comment (suo) [inline torch/csrc/jit/ir.cpp]: what happens if we're also printing source location? Doesn't this end up looking weird?

Comment (ZolotukhinM) [reply torch/csrc/jit/ir.cpp]: Yeah, we might get double comments. I traded off for code simplicity here, but I can change it if you think it's bad :)

Comment (suo) [reply torch/csrc/jit/ir.cpp]: Hm, maybe we should just make the function name an attribute of the node, for convenience? Similar to what we do with methods

Comment (suo) [reply torch/csrc/jit/ir.cpp]: Your call, both seem fine to me

Comment (ZolotukhinM) [reply torch/csrc/jit/ir.cpp]: > Hm, maybe we should just make the function name an attribute of the node, for convenience? Similar to what we do with CallMethod

That's a good idea, I'll try this version.

Comment (ZolotukhinM) [reply torch/csrc/jit/ir.cpp]: Please take a look at #24446!

**Comment by ZolotukhinM** [comment]
Closed in favor of #24446.

**Pull request closed without merging**